### DISCUSSION TITLE
Scheduling

### DISCUSSION THREAD

**Discussion by vc42** [tagyoureit/nodejs-poolController#849] [Nixie]
Hi,

My pool setup is extremely simple: I have a Intelliflo VS pump and an IC40 chlorinator.

My hope was that I would be able to use poolController mainly as a set of sensors to see salt level, pump speed, power consumption, this sort of things and feed them to Home Assistant, while letting the pump and the chlorinator run on their own. So, I got an Elfin EW11, installed poolController in the single body mode and started experimenting.

Apparently, the hoped for scenario is impossible by design because as soon as you start talking over RS485, local controls are disabled. Ok, fine, I wanted to play with the chlorinator, as the first step anyway, and see how well it goes.  I pushed the Pool button to activate the pool, set various levels of chlorine, everything worked ok.

What I noticed is that when you kill poolController, the chlorinator retains its settings, but when you start poolController again, it resets the chlorine level to zero until I push the Pool button again to make it green.

My question, therefore, is: Do I need a schedule to start pool activity ? The state of this Pool button(active) is *not* saved between restarts ?

**Comment by rstrouse** [comment]
Yes.  The chlorinator should not engage unless the filter (pool in your case) is engaged.  Set a pool schedule and it will turn on the chlorinator during that time.

**Reply by vc42** [reply]
Thanks !

Ok, playing a bit more and I have a question on the RS485 messages to the chlorinator(IC40).

1. The controller(nixie) sends "set level" messages every 3 seconds or so to the chlorinator.  Is it really needed ?  Why not send it once ?  What's the required sequence ?  A ping message and then any other request ?

**Reply by rstrouse** [reply]
You must send the set comms at regular intervals or the chlorinator will stop communication over rs485. The same is true for pumps and rs485 heaters.

**Reply by vc42** [reply]
But why is it a problem ? The chlorinator appears not to be sending unsolicited messages anyway,  As I understand, it's a simple request/response dialog. 

So, why not ping the chlorinator once (if needed), wait for a response, send the set command once wait for a response and be done with it.  Where's the catch ?

**Reply by rstrouse** [reply]
Rs485 is a master / slave communication the the chlorinator will not respond unless it it asked… repeatedly.

**Reply by vc42** [reply]
Right, I know that, but why pound the chlorinator with requests every 3 seconds ?  That I do not understand. In the capture, you can see that the controller asks, and the chlorinator responds.  Why ask repeatedly ?

**Reply by CodeSlinger69** [reply]
I believe that the chlorinator needs to receive constant message pings as a safety precaution. If say the controller (nixie or ocp) dies, then the chlorinator won’t get a “stay alive” message and will shutdown. I think the pumps have a similar safety setup.

**Reply by vc42** [reply]
The chlorinator won't shutdown, though, I checked. It will continue with the latest setting if the nexie controller drops dead.  That's why I am mystified by the continuous stream of set messages.

**Reply by rstrouse** [reply]
Yes it will.  It will release its submission to the automation.  Believe me I have been to the mountain on this.

**Reply by vc42** [reply]
Well, not in my experience.  I just checked a minute ago:  when the app is shutdown, the chlorinator happily continues to work at the last setting.  When I restart the app, it connect to the chlorinator without any problem.  Besides, as I said, the chlorinator does not submit anything on its own, it always responds to requests.

**Reply by rstrouse** [reply]
In that mode press the buttons on the cell.  The cell can change and no responses will be sent to the automation.   The way Nixie does this is identical to all Pentair automation systems.  When the automation system is in control the cell control panel should be disabled.  There are three frames sent that manage it and must be repeated or it will release.

**Reply by vc42** [reply]
That's a good point, thanks.
